src/plugins/data/server/index_patterns/fetcher/index_patterns_fetcher.ts

```diff
@@ -118,4 +124,25 @@ export class IndexPatternsFetcher {
     }
     return await getFieldCapabilities(this.elasticsearchClient, indices, metaFields);
   }
+
+  /**
+   *  Returns an index pattern list of only those index pattern strings in the given list that return indices
+   *
+   *  @param patternList string[]
+   *  @return {Promise<string[]>}
+   */
+  async validatePatternListActive(patternList: string[]) {
+    const result = await Promise.all(
+      patternList.map((pattern) =>
+        this.elasticsearchClient.count({
```

Comment:
I changed from the `_resolve/index`api to the `count` api as read only users [got 403'd](https://github.com/elastic/kibana/pull/90170/checks?check_run_id=1832802887) on the resolve. count works well: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-count.html

Reply:
Its annoying that the resolve endpoint works this way but this is a good choice.

Reply:
yes it is annoying. i would not have caught it if it weren't for that ML test!

Reply:
I also tried `ElasticsearchClient.indices.exists` and it appears read-only users get 403'd on those `indices` methods as well. I'm not sure if this has been taken into account for testing anywhere that uses the `getFieldsForTimePattern` which eventually calls `ElasticsearchClient.indices.getAlias`: https://github.com/elastic/kibana/blob/master/src/plugins/data/server/index_patterns/fetcher/lib/es_api.ts#L42

Reply:
That code needs to be removed, time pattern index patterns were dropped in 7.x.